client/client.go

```diff
@@ -324,24 +324,13 @@ func Configure(logger hclog.Logger, providerConfig interface{}) (schema.ClientMe
 		// This is a try to solve https://aws.amazon.com/premiumsupport/knowledge-center/iam-validate-access-credentials/
 		// with this https://github.com/aws/aws-sdk-go-v2/issues/515#issuecomment-607387352
 		switch {
-		case accountID != "default" && account.RoleARN != "":
-			// assume role if specified (SDK takes it from default or env var: AWS_PROFILE)
+		case account.LocalProfile != "":
 			awsCfg, err = config.LoadDefaultConfig(
 				ctx,
 				config.WithDefaultRegion(defaultRegion),
+				config.WithSharedConfigProfile(account.LocalProfile),
```

Comment:
will it still continue to work with `AWS_PROFILE` ?

Reply:
Yes, because it only overrides when `account.LocalProfile` is not an empty string